### PR TITLE
feat(FXA-2227): Phase 3 — branch_geometry primitive

### PR DESCRIPTION
**Pull request by frankyxhl**
## Summary

CHG-2227 Phase 3 — the spike-exit-gated branch geometry primitive. Pure-function ASCII geometry for forward branches with edge labels and auto-detected convergence. Strictly additive: no existing renderer touched. Phases 4-7 (renderer integration + Mermaid) build on this; flag flip + v1.8.0 release ship in Phase 8.

## What lands

New file: \`src/fx_alfred/core/branch_geometry.py\` (~210 LOC).

API surface (per CHG-2227 §"Phase 3 Primitive API Contract"):

| Type | Purpose |
|---|---|
| \`BranchRenderInput\` (frozen dataclass) | Input: parent text, siblings (BranchTarget tuple from CHG-2226), sibling texts, converges_to, box_width |
| \`BranchRenderOutput\` (frozen dataclass) | Output: lines, parent_anchor_row, convergence_anchor_row, **step_anchor_rows** (dict[str, int] — required by dag_graph for loop track placement) |
| \`render_branch()\` | Main entry point |
| \`compute_column_offsets()\` | \`c_i = box_width//2 + i*(box_width + GUTTER)\` |
| \`render_label_row()\` | Centers each label at \`c_i\` with collision-avoidance max-width |
| \`render_join_row()\` | \`└──┼──┘\` with \`┼\` at \`(c_1 + c_N) // 2\` |

## Geometry rules (per PRP-2225 §Geometry algorithm sketch)

- Each label centers at column \`c_i\` (above its \`┬\`); max-width = \`min(c_i - prev_boundary, next_boundary - c_i) * 2 - 2\` cells via \`wcwidth\`
- \`prev_boundary[i] = (c_{i-1}+c_i)//2 if i>=1 else 0\`
- \`next_boundary[i] = (c_i+c_{i+1})//2 if i<N else total_width-1\`
- Empty labels yield blank slots; all-empty collapses the row entirely
- Hard cap: 4 siblings (raises \`ValueError\` on 5+)

## Spike-exit gate cleared

| Item | Status |
|---|---|
| 11/11 invariants (I1-I11) | ✅ pass |
| 6/6 goldens (2-way, 3-way, 4-way, dangling, CJK, Audit Ledger) | ✅ pass |
| pyright src/ | ✅ 0 errors |
| ruff check + format | ✅ clean |

The Audit Ledger golden (PR #67 / Codex R4 advisory) replaces R1's manual-eyeball gate with a programmatic diff-asserting test against PRP-2225's worked example: 3-way branch with CJK labels (通过/失败/升级), CJK sibling texts (五类 No Silent / Entry 完整性 / Challenge 入口), and convergence to "Audit Ledger Entry".

## Test plan

- [x] Full suite: 755 → 775 tests (+20 new)
- [x] Lint, format, pyright clean
- [x] No imports from \`dag_graph\` or \`ascii_graph\` (invariant I9 — primitive is renderer-agnostic)
- [ ] Multi-model code review (COR-1602 + COR-1610) — to be dispatched after this PR opens

## New runtime dep

\`wcwidth>=0.2.13\` added to \`pyproject.toml\`. Required for cell-width-correct label truncation in CJK terminals (\`unicodedata.east_asian_width\` is not a substitute — Codex PR #65 R3 review explicitly endorsed). Pure-Python, MIT, ~2KB.

## After merge

CHG-2227 Phase 4 (\`dag_graph.py\` nested integration) branches off main and calls \`render_branch()\` from the primitive. \`step_row_index\` keys remain int-typed; siblings rendered as a group; no widening anywhere.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/fx_alfred/core/branch_geometry.py [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    """Render a forward branch with optional auto-detected convergence.
+
+    See module docstring + invariants I1–I11 for the contract. Raises
+    ``ValueError`` if ``len(input.siblings) > 4`` (invariant I10).
+    """
+    n = len(input.siblings)
+    if n > _MAX_SIBLINGS:
+        raise ValueError(f"branch_geometry hard cap is 4 siblings; got {n}")
+    if n == 0:
+        raise ValueError("branch_geometry requires >= 2 siblings")
```
[the remark this answers is not in the document]
Comment (frankyxhl) [reply src/fx_alfred/core/branch_geometry.py]: Adopted in commit 1dd32f0. Added `if n < 2: raise ValueError(...)` guard at the top of `render_branch`. New test `test_I10_lower_arity_cap` enforces. Same fix landed via the multi-model code review track (Codex 8.6 review) as Finding 1.

File: src/fx_alfred/core/branch_geometry.py [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+            arrow_row[off] = "▼"
+    lines.append("".join(arrow_row))
+
+    # Sibling boxes — three rows each, side-by-side.
+    sibling_box_top = list(" " * total_width)
+    sibling_box_middle = list(" " * total_width)
+    sibling_box_bottom = list(" " * total_width)
+    step_anchor_rows: dict[str, int] = {}
+    middle_row_idx = len(lines) + 1  # row where text body sits
+    for i, (target, body) in enumerate(zip(input.siblings, input.sibling_texts)):
```
[the remark this answers is not in the document]
Comment (frankyxhl) [reply src/fx_alfred/core/branch_geometry.py]: Adopted in commit 1dd32f0. Added `if len(inp.sibling_texts) != n: raise ValueError(...)` guard. New test `test_sibling_texts_length_mismatch_raises` enforces. Prevents the silent zip truncation that would have left some declared siblings without a box or anchor row, violating I11.

File: src/fx_alfred/core/branch_geometry.py [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+    them out and total visible width stays correct. Caller filters ``""``
+    via ``"".join(c for c in row if c != "")``.
+    """
+    col = start
+    for ch in text:
+        cw = wcwidth.wcwidth(ch)
+        if cw < 0:
+            cw = 1
+        if 0 <= col < len(row):
+            row[col] = ch
```
[the remark this answers is not in the document]
Comment (frankyxhl) [reply src/fx_alfred/core/branch_geometry.py:187]: Adopted in commit 59340d4. `_paint_into_row` now handles `cw == 0` (combining marks, ZWJ, variation selectors) by walking back to the previous base cell and *appending* the mark, instead of overwriting the current cell. Walks past wide-char trailing-blanks (`""`) to find the actual base. New test `test_paint_combining_marks_attach_to_base_char` verifies that "Café" written as `'C', 'a', 'f', 'e', '\\u0301'` produces `"é"` as a substring (mark adjacent to base) and preserves I2. Edge case (combining mark as first char with no base): silently dropped.
